fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/api/client/networking/v1/ClientPlayNetworking.java

```diff
@@ -62,6 +76,39 @@ public static boolean registerGlobalReceiver(Identifier channelName, PlayChannel
 		return ClientNetworkingImpl.PLAY.registerGlobalReceiver(channelName, channelHandler);
 	}
 
+	/**
+	 * Registers a handler for a packet type.
+	 * A global receiver is registered to all connections, in the present and future.
+	 *
+	 * <p>If a handler is already registered for the {@code type}, this method will return {@code false}, and no change will be made.
+	 * Use {@link #unregisterGlobalReceiver(PacketType)} to unregister the existing handler.
+	 *
+	 * @param type the packet type
+	 * @param handler the handler
+	 * @return false if a handler is already registered to the channel
+	 * @see ClientPlayNetworking#unregisterGlobalReceiver(PacketType)
+	 * @see ClientPlayNetworking#registerReceiver(PacketType, PlayPacketHandler)
+	 */
+	public static <T extends FabricPacket> boolean registerGlobalReceiver(PacketType<T> type, PlayPacketHandler<T> handler) {
+		return registerGlobalReceiver(type.getId(), new PlayChannelHandlerProxy<T>() {
+			@Override
+			public PlayPacketHandler<T> getOriginalHandler() {
+				return handler;
+			}
+
+			@Override
+			public void receive(MinecraftClient client, ClientPlayNetworkHandler networkHandler, PacketByteBuf buf, PacketSender sender) {
+				T packet = type.read(buf);
+
+				if (handler.processOnNetworkThread(client, client.player, packet, sender)) {
+					client.execute(() -> {
+						if (networkHandler.getConnection().isOpen()) handler.receive(client, client.player, packet, sender);
+					});
+				}
+			}
+		});
```

Comment:
Nit: maybe move this to an impl class, and possibly remove duplicatated logic with registerReceiver?

Reply:
Maybe, want more opinions on this

Reply:
+1 on removing duplicated logic